mne/viz/_brain/_brain.py

```diff
@@ -47,6 +47,157 @@ def safe_event(fun, *args, **kwargs):
         traceback.print_exc(file=sys.stderr)
 
 
+class _Overlay(object):
+    def __init__(self, scalars, colormap, rng, opacity):
+        self._scalars = scalars
+        self._colormap = colormap
+        self._rng = rng
+        self._opacity = opacity
+
+    def to_colors(self):
+        from matplotlib.cm import get_cmap
+        from matplotlib.colors import ListedColormap
+        if isinstance(self._colormap, str):
+            cmap = get_cmap(self._colormap)
+        else:
+            cmap = ListedColormap(self._colormap / 255.)
+
+        def diff(x):
+            return np.max(x) - np.min(x)
+
+        def norm(x, rng=None):
+            if rng is None:
+                rng = [np.min(x), np.max(x)]
+            return (x - rng[0]) / (rng[1] - rng[0])
+
+        rng = self._rng
+        scalars = self._scalars
+        if diff(scalars) != 0:
+            scalars = norm(scalars, rng)
+
+        colors = cmap(scalars)
+        if self._opacity is not None:
+            colors[:, 3] *= self._opacity
+        return colors
+
+
+class _LayeredMesh(object):
+    def __init__(self, renderer, vertices, triangles, normals):
+        self._renderer = renderer
+        self._vertices = vertices
+        self._triangles = triangles
+        self._normals = normals
+
+        self._polydata = None
+        self._actor = None
+        self._is_mapped = False
+
+        self._cache = None
+        self._overlays = dict()
+
+        self._default_scalars = np.ones(vertices.shape)
+        self._default_scalars_name = 'Data'
+
+    def map(self):
+        kwargs = {
+            "color": None,
+            "pickable": True,
+            "rgba": True,
+        }
+        mesh_data = self._renderer.mesh(
+            x=self._vertices[:, 0],
+            y=self._vertices[:, 1],
+            z=self._vertices[:, 2],
+            triangles=self._triangles,
+            normals=self._normals,
+            scalars=self._default_scalars,
+            **kwargs
+        )
+        self._actor, self._polydata = mesh_data
+        self._is_mapped = True
+
+    def _compute_over(self, B, A):
+        alpha_a = A[:, 3]
+        alpha_b = B[:, 3]
+        alpha_c = alpha_a + alpha_b * (1 - alpha_a)
+        a = A[:, :3]
+        b = B[:, :3]
+        c = (a.T * alpha_a).T + (b.T * alpha_b * (1 - alpha_a)).T
+        c = (c.T / alpha_c).T
+        return np.c_[c, alpha_c]
```

Comment:
This is equivalent but uses broadcasting and fewer copies:
```suggestion
        assert A.ndim == B.ndim == 2
        assert A.shape[1] == B.shape[1] == 4
        A_w = A[:, 3:]  # * 1
        B_w = B[:, 3:] * (1 - A_w)
        C = A.copy()
        C[:, :3] *= A_w
        C[:, :3] += B[:, :3] * B_w
        C[:, 3:] += B_w
        C[:, :3] /= C[:, 3:]
        return np.clip(C, 0, 1, out=C)
```